pyro/primitives.py

```diff
@@ -127,7 +127,8 @@ def sample(self, sample_shape=torch.Size()):
         if subsample_size == self.size:
             result = torch.LongTensor(list(range(self.size)))
         else:
-            result = torch.randperm(self.size)[:self.subsample_size]
+            # torch.randperm does not have a CUDA implementation
+            result = torch.randperm(self.size, device=torch.device('cpu'))[:self.subsample_size]
```

Comment:
if `randperm` does not have cuda implementation, thsi will throw an error either way on cuda right

Reply:
This is basically generating a random permutation on the CPU, via the `device` argument and then sending it to the GPU on the next line. The device argument is needed now, as otherwise `randperm` internally checks the device type and throws an error if it is cuda (which it will be for instance if we set the default tensor type as CUDA).